dtreeviz/trees.py

```diff
@@ -1281,11 +1330,15 @@ def draw_legend(shadow_tree, target_name, filename, colors=None, fontname="Arial
                                 facecolor=color_map[c], label=class_names[i])
         boxes.append(box)
 
+    draw_legend_frame=True
+    if colors['legend_edge'] is None:
```

Comment:
True, and this could be fixed with `colors.get('legend_edge', None)` or something similar, **but** this is also a problem with **all** of the `colors` keys. For example:
```
    ax.scatter(x_data, y_data, marker='o', alpha=colors['scatter_marker_alpha'], c=colors['scatter_marker'],
               s=markersize,
               edgecolor=colors['scatter_edge'], lw=.3)
```

Do you want to fix all of the potential KeyErrors, maybe by checking the `colors` dict once upfront, or just let KeyErrors occur if the user misses a needed key and let them sort it out?